### DISCUSSION TITLE
Using [@types/google.accounts], getting `google is not defined` in Vue.JS application

### DISCUSSION THREAD

**Discussion by Hopson97** [DefinitelyTyped/DefinitelyTyped#61725] [Issues with a @types package]
I am attempting to use the google.accounts library in a Vue.JS application, where I have

1. Installed the package using `npm i @types/google.accounts`
2. Included the typing the `tsconfig.json` file (`"types": ["google.accounts"`)
3. Added `/// <reference types="google.accounts" />` at the top of the file I that needs to use it

But doing anything with the google API eg `google.id.initialize` yields the error `google is not defined, Cannot find name 'google'`.

Any help is appreciated, thank you!

[a comment by an automated account is not shown]

**Comment by kelinwu** [comment]
yarn add @types/google.accounts

**Reply by prabnur** [reply]
`yarn add @types/google.accounts -D` is preferable

**Comment by prabnur** [comment]
In my case I suspect that the types are there but I'm not loading the scripts they are wrappers for.

**Comment by PartyLich** [comment]
Apologies for the delay. Iirc, this package as currently written only describes the `google.accounts` portion of the namespace, following the same approach as [`gapi.auth2`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/5772995e8df4423db58efb6e7220cb394d0c68ab/types/gapi.auth2/index.d.ts#L9). https://github.com/DefinitelyTyped/DefinitelyTyped/blob/5c3cb9509952b66eee587e2b53a1fbe0c1d21013/types/google.accounts/index.d.ts#L7-L8

While I haven't personally reproduced your error, possibly it spawns from not having a separate `google` namespace declared. You could try adjusting the declaration to follow something like the style in google.picker: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/2e4fed9cd4c98972dc22b066cf6d34fbf79c7b1d/types/google.picker/index.d.ts#L9-L10

If that works, you or I can open a PR and get it updated.

**Comment by thern743** [comment]
Anyone ever find a solution to this issue? Same issue in Angular (I suspect any framework that uses Typescript).

**Reply by Hopson97** [reply]
Hi Anthony, unfortunately I don't believe I did :(

**Comment by AndiDog** [comment]
You need to [load the client library](https://developers.google.com/identity/gsi/web/guides/get-google-api-clientid#load_the_client_library), not only the types. For me (ViteJS+React setup), the entry `"types": ["google.accounts"]` in `tsconfig.json` was then enough to make it work.
